cli/src/deploy/mod.rs

```diff
@@ -105,61 +105,547 @@ pub enum PostV1Error {
+      let client_email: String = Input::new()
+        .with_prompt("GCP Client Email")
+        .with_initial_text(cred.clientEmail.to_string())
+        .interact_text()
+        .unwrap();
+      let project_id: String = Input::new()
+        .with_prompt("GCP Project ID")
+        .with_initial_text(cred.projectId.to_string())
+        .interact_text()
+        .unwrap();
+      let private_key: String = Input::new()
+        .with_prompt("GCP Private Key")
+        .with_initial_text(cred.privateKey.to_string())
+        .interact_text()
+        .unwrap();
+      credentials.insert(
+        cred_name,
+        CredentialsProfile {
+          credentials: Credentials::GCP(GCPCredentials {
+            privateKey: private_key,
+            clientEmail: client_email,
+            projectId: project_id,
+          }),
+          cloudProvider: "GCP".to_owned(),
+        },
+      );
+    }
+    Credentials::Azure(cred) => {
+      let application_id: String = Input::new()
+        .with_prompt("Azure Application ID")
+        .with_initial_text(cred.applicationId.to_string())
+        .interact_text()
+        .unwrap();
+      let directory_id: String = Input::new()
+        .with_prompt("Azure Directory ID")
+        .with_initial_text(cred.directoryId.to_owned())
+        .interact_text()
+        .unwrap();
+      let subscription_id: String = Input::new()
+        .with_prompt("Azure Subscription ID")
+        .with_initial_text(cred.subscriptionId.to_string())
+        .interact_text()
+        .unwrap();
+      let secret: String = Input::new()
+        .with_prompt("Azure Secret")
+        .with_initial_text(cred.secret.to_string())
+        .interact_text()
+        .unwrap();
+      credentials.insert(
+        cred_name,
+        CredentialsProfile {
+          credentials: Credentials::Azure(AzureCredentials {
+            applicationId: application_id,
+            subscriptionId: subscription_id,
+            directoryId: directory_id,
+            secret: secret,
+          }),
+          cloudProvider: "Azure".to_owned(),
+        },
+      );
+    }
+  }
+  update_cred_file(credentials).await;
+  println!("Successfully edited \"{}\" Credential", name.to_string());
+}
+
+// prompt the user to create a deploy credential if none exists
+pub async fn prompt_new_cred() {
+  let prompt = "No Credentials have been created. Let's create one?";
+  if Confirm::new().with_prompt(prompt).interact().unwrap() {
+    add_cred().await;
+  }
+}
+
+// prompt the user to create a deploy config if none exists
+pub async fn prompt_new_config() {
+  let prompt = "No Deploy Configs have been created. Let's create one?";
+  if Confirm::new().with_prompt(prompt).interact().unwrap() {
+    add_deploy_config().await;
+  }
+}
+
+pub async fn remove_cred() {
+  let mut creds = get_creds().await;
+  let cred_options = creds.keys().cloned().collect::<Vec<String>>();
+  if cred_options.len() == 0 {
+    prompt_new_cred().await
```

Comment:
don't we have to reload the credentials? and ensure that they were actually entered?

Reply:
also: if there are no credentials, shouldn't we just state that and exit? Idk why someone would want to remove credentials they just added

Reply:
> don't we have to reload the credentials? and ensure that they were actually entered?

That's fair, the user could reject the prompt. I will end the program if that is the case here

Reply:
> also: if there are no credentials, shouldn't we just state that and exit? Idk why someone would want to remove credentials they just added

if a credential exists then we allow them to edit/remove, but I thought it would flow better if a user (accidentally?) calls this when no credentials exist if we state there are no credentials, but also ask them if they wish to create one